src/simcore_service_deployment_agent/git_url_watcher.py

```diff
@@ -31,60 +41,88 @@ async def _git_clone_repo(
     password: str = None,
 ):
     if username != None and password != None and username != "" and password != "":
-        # Black code linter crashed(!) here and the pre-commit hook failed. This three-liner is thus a dirty hack so I can commit in the first place... :(
-        cmd = "git clone -n "
-        cmd += str(URL(repository).with_user(username).with_password(password))
-        cmd += f" --depth 1 {directory} --single-branch --branch {branch}"
+        cmd = [
```

Comment:
you could also write like so:
cmd = f"git clone -n {URL(repository)}".split(" ")